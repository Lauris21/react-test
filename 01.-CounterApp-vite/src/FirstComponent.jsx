import PropTypes from "prop-types";
import React from "react";

const FirstComponent = ({ title, subtitle, name }) => {
  return (
    <div>
      <h1>FirstComponent</h1>
      <h2 data-testid="test-title">{title}</h2>
      <p>{subtitle}</p>
      <p>{name}</p>
    </div>
  );
};

FirstComponent.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  name: PropTypes.string,
};

export default FirstComponent;
