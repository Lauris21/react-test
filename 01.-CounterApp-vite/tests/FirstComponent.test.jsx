import { render } from "@testing-library/react";
import FirstComponent from "../src/FirstComponent";
import React from "react";

describe("Pruebas en <FirstComponent />", () => {
  // Hacemos una fotografÃ­a al componente
  test("debe hacer match con el snapshot", () => {
    const title = "Hola, soy tu padre";
    const { container } = render(<FirstComponent title={title} />);

    // Render aactualiza el objeto screem

    expect(container).toMatchSnapshot();
  });

  test("debe contener el title en un h2", () => {
    const title = "Hola, soy tu padre";
    const { container, getByText } = render(<FirstComponent title={title} />);

    expect(getByText(title)).toBeTruthy();

    const h2 = container.querySelector("h2");

    // expect(h2.innerHTML).toBe(title);

    expect(h2.innerHTML).toContain(title);
  });
});
