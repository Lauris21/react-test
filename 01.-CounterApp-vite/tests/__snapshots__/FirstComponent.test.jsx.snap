// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstComponent /> debe hacer match con el snapshot 1`] = `
<div>
  <div>
    <h1>
      FirstComponent
    </h1>
    <h2>
      Hola, soy tu padre
    </h2>
    <p />
    <p />
  </div>
</div>
`;
